SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialTypebindingManager.cpp

```diff
@@ -28,169 +30,207 @@ void USpatialTypebindingManager::Init(USpatialNetDriver* InNetDriver)
+		}
+	});
 
-			if (Property->PropertyFlags & CPF_Handover)
-			{
-				for (int32 ArrayIdx = 0; ArrayIdx < PropertyIt->ArrayDim; ++ArrayIdx)
-				{
-					FHandoverPropertyInfo HandoverInfo;
-					HandoverInfo.Handle = Info.HandoverProperties.Num() + 1; // 1-based index
-					HandoverInfo.Offset = Property->GetOffset_ForGC() + Property->ElementSize * ArrayIdx;
-					HandoverInfo.ArrayIdx = ArrayIdx;
-					HandoverInfo.Property = Property;
 
-					Info.HandoverProperties.Add(HandoverInfo);
-				}
-			}
+	for (auto& SubobjectDataPair : SchemaDatabase->ClassPathToSchema[Class->GetPathName()].SubobjectData)
+	{
+		int32 Offset = SubobjectDataPair.Key;
+		FSubobjectSchemaData SubobjectSchemaData = SubobjectDataPair.Value;
 
-			if (Property->PropertyFlags & CPF_AlwaysInterested)
-			{
-				for (int32 ArrayIdx = 0; ArrayIdx < PropertyIt->ArrayDim; ++ArrayIdx)
-				{
-					FInterestPropertyInfo InterestInfo;
-					InterestInfo.Offset = Property->GetOffset_ForGC() + Property->ElementSize * ArrayIdx;
-					InterestInfo.Property = Property;
+		FSoftClassPath SubobjectClassPath(SubobjectSchemaData.ClassPath);
+		UClass* SubobjectClass = SubobjectClassPath.TryLoadClass<UObject>();
+		if (SubobjectClass == nullptr)
+		{
+			continue;
+		}
 
-					Info.InterestProperties.Add(InterestInfo);
-				}
-			}
+		FClassInfo* SubobjectInfoPtr = FindClassInfoByClass(SubobjectClass);
+		if (SubobjectInfoPtr == nullptr)
+		{
+			continue;
 		}
 
+		// Make a copy of the already made FClassInfo for this specific subobject
+		TSharedPtr<FClassInfo> ActorSubobjectInfo = MakeShared<FClassInfo>(*SubobjectInfoPtr);
+		ActorSubobjectInfo->SubobjectName = SubobjectSchemaData.Name;
+
 		ForAllSchemaComponentTypes([&](ESchemaComponentType Type)
 		{
-			Worker_ComponentId ComponentId = SchemaDatabase->ClassPathToSchema[Class->GetPathName()].SchemaComponents[Type];
+			Worker_ComponentId ComponentId = SubobjectSchemaData.SchemaComponents[Type];
 			if (ComponentId != 0)
 			{
-				Info.SchemaComponents[Type] = ComponentId;
-				ComponentToClassMap.Add(ComponentId, Class);
-				ComponentToOffsetMap.Add(ComponentId, 0);
-				ComponentToCategoryMap.Add(ComponentId, (ESchemaComponentType)Type);
+				ActorSubobjectInfo->SchemaComponents[Type] = ComponentId;
+				ComponentToClassInfoMap.Emplace(ComponentId, ActorSubobjectInfo);
+				ComponentToOffsetMap.Add(ComponentId, Offset);
+				ComponentToCategoryMap.Add(ComponentId, ESchemaComponentType(Type));
 			}
 		});
 
-		Info.Class = Class;
+		Info.SubobjectInfo.Add(Offset, ActorSubobjectInfo);
+	}
+}
 
-		ClassInfoMap.Emplace(Class, Info);
+UClass* USpatialTypebindingManager::LoadClassForComponent(Worker_ComponentId ComponentId)
+{
+	// Only try to load classes for generated components
+	if (ComponentId < SpatialConstants::STARTING_GENERATED_COMPONENT_ID)
+	{
+		return nullptr;
 	}
 
-	for (UClass* Class : SupportedClasses)
+	for (auto& ObjectDataPair : SchemaDatabase->ClassPathToSchema)
```

Comment:
Nit: `ObjectDataPair` sounds like it stores something for an object instance, while this actually stores a pair of class path and class info. `ClassInfoPair`?

Reply:
I believe this is the same naming as in the previous implementation. Note that value of the pair is not class info, it's a `FSchemaData`. Do you still think it should be changed?

Reply:
You're right. I would prefer `ClassDataPair` though (my problem is that `Object` seems to imply an object instance).